stubs/CoreGenericFunctions.phpstub

```diff
@@ -484,6 +484,27 @@ function strtolower(string $str) : string {}
  */
 function strtoupper(string $str) : string {}
 
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($str, $to) -> return
+ */
+function strtr(string $str, string $from, string $to) : string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($str, $replace_pairs) -> return
+ */
+function strtr(string $str, array $replace_pairs) : string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($string, $replacement) -> return
+ */
+function substr_replace(mixed $string, mixed $replacement, mixed $start, ?mixed $length) : mixed {}
```

Comment:
TODO: Fix the mixed.

Reply:
It would be really nice if conditional types for parameters would be supported. They don't seem to be at the moment.

Example:

```
 * @param ($string is array<string> ? array<string> : string) $replacement
 * @param ($string is array<string> ? array<int> : int) $start
 * @param ($string is array<string> ? array<int> : int) $length
```

Reply:
I also need to read up on `FunctionReturnTypeProviderInterface` / `StrReplaceReturnTypeProvider` to understand how one propagates taint there.

Reply:
> It would be really nice if conditional types for parameters would be supported. They don't seem to be at the moment.

Yep. Also related: #698

Reply:
@LukasReschke taints are propagated by copying the `parent_nodes` property on a type.

As an aside it's slightly weird to have so much information on a type object – I may migrate some of it to a `Data` object or similar, but that'd be a major version breaking change.